docs/configuration/admin-config.md
# Admin Config

## General Notes on Admin Config

Modifying the admin configuration is a real time operation that directly effects the operation of your Janus server.
Be careful when changing settings, especially the security ones.

## Security

Janus supports both a built-in authentication mechanism based on email and password as well as single sign on via OAuth2.  The built-in authentication mechanism is enabled by default.

It is highly recommended that you leave the built-in authentication enabled until you have thoroughly tested your SSO config,
disabling the built-in authentication could lock you out of the system if your SSO config is not correct.  To help with this
Janus supports running both configs at the same time.

Admin configs are managed at: `http://<janus-host>/admin` and requires the global admin permission.

| Config Item            | Explanation                                                               |
|------------------------|---------------------------------------------------------------------------|
| Disable Password Login | Removes the ability to authenticate on the login page with email/password |
| Enable SSO             | Adds the SSO login option to the login screen                             |
| SSO Provider           | Your SSO provider, either entra (azuread), google or generic OIDC         |
| SSO Client ID          | Client ID from your identity provider                                     |
| SSO Client Secret      | Client secrets from your identity provider                                |
| Entra Tenant ID        | Only needed for entra, your azure tenant ID                               |
| Discovery URL          | Only neede for generic OIDC, discovery URL                                |

## SMTP

Janus notification channels share a common configuration for SMTP.  This configuration is used to send email notifications on job status updates.

| Config Item | Explanation                                             |
|-------------|---------------------------------------------------------|
| SMTP Server | The SMTP server to use for sending email                |
| SMTP Port   | The port on the SMTP server to use                      |
| SMTP User   | The username to use for SMTP authentication             |
| SMTP Pass   | The password to use for SMTP authentication             |
| SMTP From   | The email address to use as the from address for emails |

## Job Settings

Job settings covers general job settings that impact all jobs.

| Config Item               | Explanation                                                                                               |
|---------------------------|-----------------------------------------------------------------------------------------------------------|
| Default Timeout (Seconds) | When a user creates a new job this is the default value set unless they change it                         |
| Max Concurrent Jobs       | The max number of jobs that can be running on the Janus server, any job started over this limit will fail |

## Data Retention

Data retention settings control how long Janus keeps job history and job output.

| Config Item      | Explanation                                                                                         |
|------------------|-----------------------------------------------------------------------------------------------------|
| Job History Days | The number of days to keep job history in the database, older history will be automatically deleted |

docs/configuration/runtime-config.md
# Runtime Config

Janus is designed to be simple to get started with so there are no config files and the list of configuration
options are limited.  All configuration items are set by environment variables.  Settings these can vary based on your
environment.

| Variable              | Explanation                                              | Default Value         |
|-----------------------|----------------------------------------------------------|-----------------------|
| JANUS_PORT            | Port Janus runs on                                       | 8080                  |
| JANUS_DB_TYPE         | Database type (sqlite or postgres)                       | sqlite                |
| JANUS_DB_PATH         | Path to sqlite DB file, only used when DB_TYPE is sqlite | janus.db              |
| JANUS_DB_HOSTNAME     | Postgres database hostname                               | localhost             |
| JANUS_DB_PORT         | Postgres database port                                   | 5432                  |
| JANUS_DB_NAME         | Postgres database name                                   | postgres              |
| JANUS_DB_USER         | Postgres database username                               | postgres              |
| JANUS_DB_PASSWORD     | Postgres database password                               | postgres              |
| JANUS_DB_SSL_MODE     | Postgres SSL connection mode                             | disable               |
| JANUS_URL             | URL to access janus on, should be set to your domain     | http://localhost:8080 |
| JANUS_DISABLE_METRICS | Disable prometheus metric listener                       | false                 |
| JANUS_METRICS_PORT    | Port for prometheus metrics                              | 8081                  |
| JANUS_SESSION_NAME    | Name of the cookie for authentication tracking           | janus                 |
| JANUS_BRAND_NAME      | Controls the site name and brand name                    | Janus                 |

pkg/configuration.go
package pkg

import (
	"github.com/kelseyhightower/envconfig"
	_ "github.com/lib/pq"
	_ "github.com/mattn/go-sqlite3"
)

type Config struct {
	Port            int    `envconfig:"JANUS_PORT" default:"8080"`
	DBType          string `envconfig:"JANUS_DB_TYPE" default:"sqlite"`
	DBPath          string `envconfig:"JANUS_DB_PATH" default:"janus.db"`
	DBHostname      string `envconfig:"JANUS_DB_HOSTNAME" default:"localhost"`
	DBPort          int    `envconfig:"JANUS_DB_PORT" default:"5432"`
	DBName          string `envconfig:"JANUS_DB_NAME" default:"postgres"`
	DBUser          string `envconfig:"JANUS_DB_USER" default:"postgres"`
	DBPassword      string `envconfig:"JANUS_DB_PASSWORD" default:"postgres"`
	DBSSLMode       string `envconfig:"JANUS_DB_SSL_MODE" default:"disable"`
	ServerURL       string `envconfig:"JANUS_URL" default:"http://localhost:8080"`
	DevelopmentMode bool   `envconfig:"JANUS_DEVELOPMENT_MODE" default:"false"`
	DisableMetrics  bool   `envconfig:"JANUS_DISABLE_METRICS" default:"false"`
	MetricsPort     int    `envconfig:"JANUS_METRICS_PORT" default:"8081"`
	SessionName     string `envconfig:"JANUS_SESSION_NAME" default:"janus"`
	BrandName       string `envconfig:"JANUS_BRAND_NAME" default:"Janus"`
}

func LoadConfig() (*Config, error) {
	var c Config
	err := envconfig.Process("janus", &c)
	if err != nil {
		return nil, err
	}
	return &c, nil
}
